### PR TITLE
User Story 5

### LINKED ISSUE #8: User Story 5, Admin Navigation
As an admin
I see the same links as a regular user
Plus the following links
- a link to my admin dashboard ("/admin")
- a link to see all users ("/admin/users")

Minus the following links/info
- a link to my shopping cart ("/cart") or count of cart items

### PR DESCRIPTION
**Pull request by brian-greeson**
# Admin specific links appear in the navbar when admin is logged in
## Created
- dashboard_controller.rb
  - Placeholder for route to work
- admin/user_controller.rb
 - Placeholder for route to work
## Modified
- application_controller
  - added current_admin? method to helper methods
  - added helper methods
    - current_admin? : returns true if current logged in user has admin role
    - requier_admin : renders 404 if current logged in user is NOT admin 
- session_controller.rb
  - added placeholder destroy action with redirect to root 
- application.html.erb 
  - Updated links available to admin users
    - cart
    - login
    - logout
    - register
    - dashboard
    - profile
    - users
- routes.rb
  - added namespace for mechant and admin
  - logout route
- admin/nav_spec.rb
   - Added test for user story 5 features
- visitors/nav_spec.rb
   - Updated test to be in line with pr #62

### CODE REVIEW

File: spec/features/admin/nav_spec.rb [hunk 1]
```diff
@@ -0,0 +1,87 @@
+require 'spec_helper'
+require 'rails_helper'
+
+RSpec.describe 'As a Admin I have a navbar', type: :feature do
```
Comment (kelshadarby) [inline spec/features/admin/nav_spec.rb:4]: Nice naming of this test

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -11,4 +11,12 @@ def current_user
     @current_user ||= User.find(session[:user_id]) if session[:user_id]
   end
 
+  def current_admin?
+    current_user && current_user.admin?
+  end
+
+  def require_admin
```
Comment (fredrondina) [inline app/controllers/application_controller.rb:18]: nice job setting up this check method

File: config/routes.rb [hunk 3]
```diff
@@ -7,6 +7,7 @@
 
   namespace :admin do
```
Comment (kelshadarby) [inline config/routes.rb:8]: This looks good!

File: app/views/layouts/application.html.erb [hunk 4]
```diff
@@ -13,12 +13,17 @@
       <%= link_to "Home", "/" %>
       <%= link_to "All Merchants", "/merchants"%>
       <%= link_to "All Items", "/items"%>
-      <%= link_to "Cart: #{cart.total_items}", "/cart" %>
+      <%= link_to "Cart: #{cart.total_items}", "/cart" if !current_admin? %>
       <%= link_to "Register", register_path if !current_user %>
       <%= link_to "Login", login_path if !current_user %>
-      <%= link_to "View My Profile", "/profile" if current_user %>
+      <%= link_to "Profile", profile_path if current_user %>
```
Comment (fredrondina) [inline app/views/layouts/application.html.erb:19]: nice job with the reorganization here, will display the nav bar much more intuitively

File: app/views/layouts/application.html.erb [hunk 5]
```diff
@@ -13,12 +13,17 @@
...
+      
+      <%= link_to "Dashboard", admin_dashboard_path if current_admin? %>
+      <%= link_to "Users", admin_users_path if current_admin? %>
+  
       <%= link_to "Dashboard", merchant_dashboard_path if current_user && current_user.role == "merchant" %>
```
Comment (kelshadarby) [inline app/views/layouts/application.html.erb:24]: Note to change this to a model method

**Review by kelshadarby — APPROVED** [review]

**Review by fredrondina — APPROVED** [review]
great work, very clean

**Pull request merged**